hedera-mirror-rest/utils.js

```diff
@@ -713,8 +713,14 @@ const randomString = async (length) => {
   return bytes.toString('hex');
 };
 
+const hexPrefix = '0x';
 const addHexPrefix = (hexString) => {
-  return `0x${hexString}`;
+  if (_.isEmpty(hexString)) {
+    return null;
+  }
+
+  hexString = `${hexString}`;
+  return hexString.substring(0, 2) === hexPrefix ? hexString : `${hexPrefix}${hexString}`;
```

Comment:
Actually in benchmarking i think the best candidate here would be `indexOf`
```suggestion
  return str.indexOf(hexPrefix) === 0 ? hexString : `${hexPrefix}${hexString}`;
```

Reply:
I don't see any benchmarks with proof of that. Even if there were a difference it would be so minute that it's not worth sacrificing the readability of using `startsWith`.

Reply:
Did a benchmark on measurethat.net/Benchmarks, `indexOf` is almost twice as fast as substring.
Not sure how it substantially reads worse. Also that can be addressed with a comment.

With high contract result usage expected with more Smart Contract scenarios as well as Staking in the future this function will see more traction.
I'd still advise we go with `indexOf`

Reply:
> `indexOf` is almost twice as fast as substring.

Umm, but that wasn't the debate. It was `indexOf` vs `startsWith`.

Reply:
`StartsWith` is a little better than `Regex` but worse than `substring`
From the bench mark i tried `indexOf` is roughly an order of magnitude faster than `startsWith`
[js-regex-vs-startswith-vs-indexof-vs-endswith-vs-charat](https://www.measurethat.net/Benchmarks/Show/12312/0/js-regex-vs-startswith-vs-indexof-vs-endswith-vs-charat)

Reply:
I have updated it to use `indexOf`.

Reply:
a bit late... it looks to me `indexOf` definitely has perf penalty on large strings without `0x` at all. 

also the benchmark linked is run on browser, we use nodejs, the numbers can be completely different.

Reply:
I ran a different benchmark locally that I customized based upon a [gist](https://gist.github.com/tabarra/d03dbcb78ce5db4c1cf02b659be688bb). It does show that `indexOf` is a lot faster than `startsWith`. For whatever reason, nodejs or v8's implementation of `startsWith` is not optimal as one would expect. But it also shows that `substring` is faster than `indexOf`. 😆  I say it's fine as `indexOf`.

```
$ node startswith.js
indexOf x 931,496,738 ops/sec ±0.92% (95 runs sampled)
startsWith x 76,676,989 ops/sec ±0.45% (94 runs sampled)
lastIndexOf x 40,956,896 ops/sec ±0.91% (94 runs sampled)
substring x 938,148,086 ops/sec ±0.52% (96 runs sampled)
slice x 930,841,502 ops/sec ±0.87% (94 runs sampled)
regex x 48,709,553 ops/sec ±1.28% (93 runs sampled)
compiled regex x 59,707,114 ops/sec ±0.27% (98 runs sampled)
>Fastest is substring,indexOf,slice
```

Reply:
Haha, you edited your comment later with almost the same comment as me using the same benchmark.

Reply:
Haha wow. Let's go with the performant option for our code base.
I'm fine with `substring` in that case :)

Reply:
It's okay as is. We'll cross any performance issues later and we got a good discussion going.